src/modules/post/post.controller.ts

```diff
@@ -3,19 +3,25 @@ import {
   Controller,
   Delete,
   Get,
+  Inject,
   Param,
   Post,
   Put,
 } from '@nestjs/common';
 import { Post as PostModel } from '@prisma/client';
 import { ApiTags } from '@nestjs/swagger';
 
-import { PostService } from './post.service';
+import { InterfaceServiceAlias } from '../../shared/constants/service.constants';
+
+import { PostInterface } from './post.interface';
 
 @ApiTags('posts')
 @Controller('/posts')
 export class PostController {
-  constructor(private postService: PostService) {}
+  constructor(
+    @Inject(InterfaceServiceAlias.POST_SERVICE)
+    private postService: PostInterface,
```

Comment:
What's the benefit of 
```
@Inject(InterfaceServiceAlias.POST_SERVICE)
```
Isn't it enough just using this
```
private postService: PostInterface,
```
?

Reply:
1. @Inject(InterfaceServiceAlias.POST_SERVICE) as the name, This is a string name of provide you want injects. I am using interface instead of class, so it is necessary.
2. Yes, and you can interest main methods in interface.